fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/navbar/NavigationBarButtons.kt

```diff
@@ -0,0 +1,206 @@
+     * and returns a tailored list of [NavBarButton]s suitable for that location. Each [NavBarButton]
+     * is configured based on the specific requirements and characteristics of the navigation target,
+     * ensuring a context-appropriate user interface.
+     */
+    fun build(location: NavBarLocation): List<NavBarButton> {
+        return when (location) {
+            is NavBarLocation.BrowserFragment -> buildNavButtonsForOpenTab(location)
+            is NavBarLocation.HomeFragment -> buildNavButtonsForHomePage(location)
+        }
+    }
+
+    private fun buildNavButtonsForOpenTab(location: NavBarLocation.BrowserFragment): List<NavBarButton> {
+        return listOf(
+            DefaultButtons.back.copy().apply {
+                onClick = {
+                    location.interactor.onBrowserToolbarMenuItemTapped(
+                        ToolbarMenu.Item.Back(viewHistory = false),
+                    )
+                }
+                onLongPress = {
+                    location.interactor.onBrowserToolbarMenuItemTapped(
+                        ToolbarMenu.Item.Back(viewHistory = true),
+                    )
+                }
+            },
+            DefaultButtons.forward.copy().apply {
+                onClick = {
+                    location.interactor.onBrowserToolbarMenuItemTapped(
+                        ToolbarMenu.Item.Forward(viewHistory = false),
+                    )
+                }
+                onLongPress = {
+                    location.interactor.onBrowserToolbarMenuItemTapped(
+                        ToolbarMenu.Item.Forward(viewHistory = true),
+                    )
+                }
+            },
+            DefaultButtons.home.copy().apply {
+                onClick = {
+                    location.navController.navigate(
+                        BrowserFragmentDirections.actionGlobalHome(),
+                    )
+                }
+            },
+            NavBarButton.TabCounterButton.apply {
+                onClick = {
+                    location.thumbnailsFeature.get()?.requestScreenshot()
+                    location.navController.nav(
+                        R.id.browserFragment,
+                        BrowserFragmentDirections.actionGlobalTabsTrayFragment(
+                            page = when (location.browsingModeManager.mode) {
+                                BrowsingMode.Normal -> Page.NormalTabs
+                                BrowsingMode.Private -> Page.PrivateTabs
+                            },
+                        ),
+                    )
+                }
+            },
+            NavBarButton.ThreeDotMenuButton(location.menuButton),
+        )
+    }
+
+    private fun buildNavButtonsForHomePage(location: NavBarLocation.HomeFragment): List<NavBarButton> {
+        return listOf(
+            DefaultButtons.back.copy(),
+            DefaultButtons.forward.copy(),
+            DefaultButtons.search.copy(),
```

Comment:
We are to add the search functionality?

Reply:
the button will be focusing the address bar, bringing in the search dialog fragment.

my understanding is, keeping the home button on the home fragment didn't make sense, so it was replaces by a button that is essentially duplicating a click on the address bar. you can see it in the general flow [section](https://www.figma.com/file/8fsCvtnweBkWGZ9NfL907l/Toolbar-Redesign?type=design&node-id=5503-276255&mode=design&t=wwdfRA9QCvjSrWAB-4)